File: go/cnpj.go
package main

// FonteData - Consultar CNPJ em Go
// Docs: https://fontedata.com/docs

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"strings"
	"time"
)

const apiKey  = "fd_live_SUA_CHAVE"
const baseURL = "https://app.fontedata.com/api/v1/consulta"

func consultarCNPJ(cnpj string) (map[string]interface{}, error) {
	cnpj = strings.NewReplacer(".", "", "/", "", "-", "").Replace(cnpj)
	url := fmt.Sprintf("%s/consulta-cnpj-receita/%s", baseURL, cnpj)

	client := &http.Client{Timeout: 30 * time.Second}
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("X-API-Key", apiKey)

	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	fmt.Printf("Custo: R$ %s | Saldo: R$ %s\n",
		resp.Header.Get("X-Request-Cost"),
		resp.Header.Get("X-Balance-Remaining"))

	body, _ := io.ReadAll(resp.Body)
	if resp.StatusCode != 200 {
		return nil, fmt.Errorf("HTTP %d: %s", resp.StatusCode, body)
	}

	var result map[string]interface{}
	if err := json.Unmarshal(body, &result); err != nil {
		return nil, err
	}
	return result, nil
}

func main() {
	cnpj := "00000000000191"
	if len(os.Args) > 1 {
		cnpj = os.Args[1]
	}

	dados, err := consultarCNPJ(cnpj)
	if err != nil {
		fmt.Fprintln(os.Stderr, "Erro:", err)
		os.Exit(1)
	}

	fmt.Printf("Razao social: %v\n", dados["razao_social"])
	fmt.Printf("Situacao:     %v\n", dados["situacao_cadastral"])
	fmt.Printf("CNAE:         %v\n", dados["cnae_fiscal_descricao"])
}

File: nodejs/cnpj.js
// FonteData - Consultar CNPJ em Node.js
// Docs: https://fontedata.com/docs

const API_KEY = 'fd_live_SUA_CHAVE';
const BASE_URL = 'https://app.fontedata.com/api/v1/consulta';

async function consultarCNPJ(cnpj) {
  const cnpjLimpo = cnpj.replace(/[\.\-\/]/g, '');
  const res = await fetch(`${BASE_URL}/consulta-cnpj-receita/${cnpjLimpo}`, {
    headers: { 'X-API-Key': API_KEY }
  });
  console.log(`Custo: R$ ${res.headers.get('x-request-cost')} | Saldo: R$ ${res.headers.get('x-balance-remaining')}`);
  if (!res.ok) throw new Error(`Erro ${res.status}`);
  return res.json();
}

const cnpj = process.argv[2] || '00000000000191';
consultarCNPJ(cnpj).then(d => {
  console.log('Razao social:', d.razao_social);
  console.log('Situacao:    ', d.situacao_cadastral);
}).catch(console.error);
